src/rgw/rgw_data_sync.cc

```diff
@@ -1249,6 +1249,24 @@ struct rgw_sync_pipe_info_set {
   }
 };
 
+class LatencyConcurrencyControl : public LatencyMonitor {
+public:
+  CephContext* const cct;
+
+  LatencyConcurrencyControl(CephContext* cct)
+    : cct(cct)  {}
+  uint32_t get_concurrency() {
+    if (avg_latency() >= 10s) [[unlikely]] {
```

Comment:
maybe these `10s` and `20s` values should scale based on `rgw_sync_lease_period`, in case we want to test with longer leases?

Reply:
Done.

Threshold is lease period / 12, and the other is just 2 * threshold.